generic/piranha-tree-sitter/src/cleanup_rules/java/scope_config.toml

```diff
@@ -0,0 +1,224 @@
+                    (formal_parameter type:(_) @r3)
+                    (formal_parameter type:(_) @r4))) @qd)
+(#eq? @z "@n")
+(#eq? @r0 "@t0")
+(#eq? @r1 "@t1")
+(#eq? @r2 "@t2")
+(#eq? @r3 "@t3")
+(#eq? @r4 "@t4")                    
+)"""
+
+# Scope matcher-generator for Java method with 4 parameters.
+[[scopes.rules]]
+matcher =  """
+((method_declaration 
+			name : (_) @n
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @t0)
+                    (formal_parameter type:(_) @t1)
+                    (formal_parameter type:(_) @t2)
+                    (formal_parameter type:(_) @t3))) @xd5)"""
+generator = """
+(((method_declaration 
+			name : (_) @z
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @r0)
+                    (formal_parameter type:(_) @r1)
+                    (formal_parameter type:(_) @r2)
+                    (formal_parameter type:(_) @r3))) @qd)
+(#eq? @z "@n")
+(#eq? @r0 "@t0")
+(#eq? @r1 "@t1")
+(#eq? @r2 "@t2")
+(#eq? @r3 "@t3")
+)"""
+
+# Scope matcher-generator for Java method with 3 parameters.
+[[scopes.rules]]
+matcher =  """
+((method_declaration 
+			name : (_) @n
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @t0)
+                    (formal_parameter type:(_) @t1)
+                    (formal_parameter type:(_) @t2))) @xd2)"""
+generator = """
+(((method_declaration 
+			name : (_) @z
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @r0)
+                    (formal_parameter type:(_) @r1)
+                    (formal_parameter type:(_) @r2))) @qd)
+(#eq? @z "@n")
+(#eq? @r0 "@t0")
+(#eq? @r1 "@t1")
+(#eq? @r2 "@t2")
+)"""
+
+# Scope matcher-generator for Java method with 2 parameters.
+[[scopes.rules]]
+matcher =  """
+((method_declaration 
+			name : (_) @n
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @t0)
+                    (formal_parameter type:(_) @t1))) @xd3)"""
+generator = """
+(((method_declaration 
+			name : (_) @z
+            parameters : (formal_parameters
+            		(formal_parameter type:(_) @r0)
+                    (formal_parameter type:(_) @r1))) @qd)
+(#eq? @z "@n")
+(#eq? @r0 "@t0")
+(#eq? @r1 "@t1")
+)"""
+
+# Scope matcher-generator for Java method with 1 parameters.
+[[scopes.rules]]
+matcher =  """
+(
+(((method_declaration 
+			name : (_) @n
+            parameters : 
+            	((formal_parameters
+            		(((formal_parameter type: (_) @t0) )? @f )) @zz)
+                )
+    @xd6)
+)
+(#not-match? @zz ",")
+(#not-eq? @zz "()")
+)"""
+generator = """
+(
+(((method_declaration 
+			name : (_) @z
+            parameters : 
+            	((formal_parameters
+            		(((formal_parameter type: (_) @r1) ) @p )) @j)
+                )
+    @4)
+)
+(#not-match? @j ",")
+(#not-eq? @j "()")
+(#eq? @z "@n")
+(#eq? @r1 "@t0")
```

Comment:
Let us add a TODO to make this scope generator be scripted instead of hardcoding the parameter count. Also, document somewhere saying that this Piranha cleanup may not work for method bodies with more than 5 parameters?